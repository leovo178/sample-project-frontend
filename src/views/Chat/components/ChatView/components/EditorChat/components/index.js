export { default as SendReservationDialog } from './SendReservationDialog'
