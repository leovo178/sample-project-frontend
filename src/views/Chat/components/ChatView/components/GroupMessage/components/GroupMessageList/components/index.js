export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
export { default as FileMessage } from './FileMessage'
export { default as TextMessage } from './TextMessage'
export { default as ReservationMessage } from './ReservationMessage'
