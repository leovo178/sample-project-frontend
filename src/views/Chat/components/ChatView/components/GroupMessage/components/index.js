export { default as GroupBreak } from './GroupBreak'
export { default as GroupMessageList } from './GroupMessageList'
