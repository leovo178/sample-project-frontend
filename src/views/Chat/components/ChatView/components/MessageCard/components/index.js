export { default as FileMessage } from './FileMessage'
export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
