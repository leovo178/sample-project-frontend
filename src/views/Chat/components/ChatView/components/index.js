export { default as EditorChat } from './EditorChat'
export { default as GroupMessage } from './GroupMessage'
export { default as Header } from './Header'
export { default as MessageCard } from './MessageCard'
export { default as PickerEmoji } from './PickerEmoji'
export { default as ViewMessage } from './ViewMessage'
export { default as MessageReservation } from './MessageReservation'
