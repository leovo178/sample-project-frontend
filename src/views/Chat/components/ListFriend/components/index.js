// export { default as FriendTag } from './components/FriendTag'
export { default as Header } from './components/Header'
export { default as ListFriendTag } from './components/ListFriendTag'
