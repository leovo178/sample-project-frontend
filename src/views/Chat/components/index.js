export { default as ListFriend } from './components/ListFriend'
export { default as ChatView } from './components/ChatView'
