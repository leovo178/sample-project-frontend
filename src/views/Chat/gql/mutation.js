import gql from 'graphql-tag'

const SET_SELECTED_USER_OF_CHAT = gql`
	mutation SetSelectedUserOfChat($selectedUser: User!) {
		setSelectedUserOfChat(selectedUser: $selectedUser) @client {
			id
			name
		}
	}
`
const SET_DRAFT_LIST = gql`
	mutation SetDraftList($draft: Draft!) {
		setDraftList(draft: $draft) @client {
			toInterestId
			message
		}
	}
`
const SET_STATUS_READED_MESSAGE = gql`
	mutation SetStatusReadedMessage($readedMessage: ReadedMessage!) {
		setStatusReadedMessage(readedMessage: $readedMessage) @client {
			fromInterestedId
			numberNoti
			lastMessage
		}
	}
`

const SET_USER_SEARCH_TEXT = gql`
	mutation SetUserSearchValueOfChat($searchValue: String!) {
		setUserSearchValueOfChat(searchValue: $searchValue) @client
	}
`

const CREATE_ZALO_MESSAGE = gql`
	mutation CreateZaloMessage($message: CreateZaloMessageInput!) {
		createZaloMessage(message: $message) {
			from {
				displayName
				id
				avatar
			}
			to {
				displayName
				id
				avatar
			}
			content
			timestamp
			id
		}
	}
`

const CREATE_ZALO_MESSAGE_ATTACHMENT = gql`
	mutation CreateZaloMessageAttachment(
		$file: CreateZaloMessageAttachmentInput!
	) {
		createZaloMessageAttachment(message: $file) {
			from {
				displayName
				id
				avatar
			}
			to {
				displayName
				id
				avatar
			}
			content
			timestamp
			id
			attachments {
				type
				payload {
					thumbnail
					description
					url
				}
			}
		}
	}
`

const SET_CREATE_ZALO_MESSAGE_ATTACHMENT = gql`
	mutation SetCreateZaloMessageAttachment($message: AttachmentMessage!) {
		setCreateZaloMessageAttachment(message: $message) @client
	}
`

export {
	SET_SELECTED_USER_OF_CHAT,
	SET_DRAFT_LIST,
	SET_STATUS_READED_MESSAGE,
	SET_USER_SEARCH_TEXT,
	CREATE_ZALO_MESSAGE,
	CREATE_ZALO_MESSAGE_ATTACHMENT,
	SET_CREATE_ZALO_MESSAGE_ATTACHMENT,
}
