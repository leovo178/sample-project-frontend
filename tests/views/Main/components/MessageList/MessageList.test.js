import React from 'react'
import { findDOMNode } from 'react-dom'

import { act, fireEvent, waitFor } from '@testing-library/react'
import { MockedProvider } from '@apollo/react-testing'
import waait from 'waait'

import { MessageList } from '@views/Main/components'

import {
	mockUserList,
	mockMessageList,
	renderDOMNode,
	getMarkup,
} from '@tests/shares/utils'
import { FETCH_USER_LIST } from '@views/User/gql/query'
import { MESSAGE_LIST } from '@views/Message/gql/query'
import { DELETE_MESSAGE, UPDATE_MESSAGE } from '@views/Message/gql/mutation'
import { PAGE_LIMIT } from '@src/configs.local'

const mocks = [
	{
		request: {
			query: FETCH_USER_LIST,
			variables: {
				query: { limit: PAGE_LIMIT },
			},
		},
		result: {
			data: {
				userList: {
					items: mockUserList,
					hasNext: true,
				},
			},
		},
	},
	{
		request: {
			query: MESSAGE_LIST,
			variables: {
				query: { userId: mockUserList[0].id, limit: PAGE_LIMIT },
			},
		},
		result: {
			data: {
				messageList: {
					items: mockMessageList,
					hasNext: true,
					total: 30,
				},
			},
		},
	},
	{
		request: {
			query: DELETE_MESSAGE,
			variables: {
				id: mockUserList[0].id,
			},
		},
		result: {
			data: {
				deleteMessage: mockMessageList[0],
			},
		},
	},
]

const resolvers = {
	Query: {
		userSearchValueOfMain: () => {
			return ''
		},
		selectedUserOfMain: () => {
			return mockUserList[0]
		},
	},
	Mutation: {
		setUserSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValueOfMain: searchValue,
				},
			})
			return searchValue
		},
	},
}

const openModifyDialog = async rendered => {
	const row = rendered.querySelectorAll(
		`[data-testid=row-${mockMessageList[0].id}]`
	)[0]

	await act(async () => {
		await row.click()
	})

	const modifyDialogTitle = document.querySelectorAll(
		`[data-testid=modifydialog-title]`
	)[0]

	expect(modifyDialogTitle).toBeTruthy()
}

const openDeleteDialog = async rendered => {
	const closeIcon = rendered.querySelectorAll(
		`[data-testid=row-closeicon-${mockMessageList[0].id}]`
	)[0]

	fireEvent.click(closeIcon)

	const deleteDialogTitle = document.querySelectorAll(
		`[data-testid=deletedialog-title]`
	)[0]

	expect(deleteDialogTitle).toBeTruthy()
}

const modifyMessage = async message => {
	const input = document.querySelectorAll(`[placeholder=placeholder]`)[0]
	const agreeButton = document.querySelectorAll(
		'[data-testid=modifydialog-agreebutton]'
	)[0]

	await act(async () => {
		fireEvent.change(input, { target: { value: message } })
	})

	expect(input.value).toBe(message)

	await act(async () => {
		await agreeButton.click()
	})
}

describe('MessageList', async () => {
	it('should match snapshot', async () => {
		const rendered = findDOMNode(
			renderDOMNode(
				getMarkup(
					<MockedProvider
						mocks={mocks}
						addTypename={false}
						resolvers={resolvers}
					>
						<MessageList />
					</MockedProvider>
				)
			)
		)
		await act(async () => {
			await waait(10)
		})

		expect(rendered).toMatchSnapshot()
	})

	describe('ModifyDialog', () => {
		let rendered

		beforeEach(() => {
			rendered = findDOMNode(
				renderDOMNode(
					getMarkup(
						<MockedProvider
							mocks={mocks}
							addTypename={false}
							resolvers={resolvers}
						>
							<MessageList />
						</MockedProvider>
					)
				)
			)
		})
		it('should allow to click No button to close dialog', async () => {
			await openModifyDialog(rendered)

			const disagreeButton = document.querySelectorAll(
				`[data-testid=modifydialog-disagreebutton]`
			)[0]

			await act(async () => {
				await disagreeButton.click()
			})

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=modifydialog-title]`)[0]
				).toBeFalsy()
			})
		})

		it('should allow to modify message', async () => {
			await waait(10)
			await openModifyDialog(rendered)
			await modifyMessage('nvdai')

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=modifydialog-title]`)[0]
				).toBeFalsy()
			})
		})

		it('should allow to submit a empty message', async () => {
			jest
				.spyOn(window, 'alert')
				.mockImplementation(
					() =>
						'Error: GraphQL error: "message.content" is not allowed to be empty'
				)
			await waait(10)
			await openModifyDialog(rendered)
			await modifyMessage('')

			expect(window.alert).toHaveBeenCalled()

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=modifydialog-title]`)[0]
				).toBeFalsy()
			})
		})

		it('should allow to click MuiBackdrop-root to close dialog', async () => {
			await waait(10)
			await openModifyDialog(rendered)

			const MuiBackdrop = document.querySelectorAll(`.MuiBackdrop-root`)[0]

			await act(async () => {
				MuiBackdrop.click()
			})

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=modifydialog-title]`)[0]
				).toBeFalsy()
			})
		})
	})

	describe('DeleteDialog', () => {
		let rendered

		beforeEach(() => {
			rendered = findDOMNode(
				renderDOMNode(
					getMarkup(
						<MockedProvider
							mocks={mocks}
							addTypename={false}
							resolvers={resolvers}
						>
							<MessageList />
						</MockedProvider>
					)
				)
			)
		})
		it('allow to click Yes button', async () => {
			jest
				.spyOn(window, 'alert')
				.mockImplementation(() => 'Error: GraphQL error')
			await waait(10)
			await openDeleteDialog(rendered)

			const agreeButton = document.querySelectorAll(
				`[data-testid=deletedialog-agreebutton]`
			)[0]

			await act(async () => {
				agreeButton.click()
			})

			expect(window.alert).toHaveBeenCalled()

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=deletedialog-title]`)[0]
				).toBeFalsy()
			})
		})

		it('allow to click No button', async () => {
			await waait(10)
			await openDeleteDialog(rendered)

			const disagreeButton = document.querySelectorAll(
				`[data-testid=deletedialog-disagreebutton]`
			)[0]

			await act(async () => {
				disagreeButton.click()
			})

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=deletedialog-title]`)[0]
				).toBeFalsy()
			})
		})
		it('allow to click MuiBackdrop-root to close dialog', async () => {
			await waait(10)
			await openDeleteDialog(rendered)

			const MuiBackdrop = document.querySelectorAll(`.MuiBackdrop-root`)[0]

			await act(async () => {
				MuiBackdrop.click()
			})

			await waitFor(() => {
				expect(
					document.querySelectorAll(`[data-testid=modifydialog-title]`)[0]
				).toBeFalsy()
			})
		})
	})
})
