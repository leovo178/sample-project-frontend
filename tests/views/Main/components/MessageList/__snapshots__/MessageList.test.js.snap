// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should match snapshot 1`] = `
<div
  style="width: 300px; height: 700px;"
>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-root-1"
  >
    <h5
      class="MuiTypography-root makeStyles-listtitle-2 MuiTypography-h5"
    >
      Total 
      30
    </h5>
    <div
      class="MuiBox-root MuiBox-root-93 makeStyles-root-90"
      data-testid="infinitetable"
    >
      <div
        class="MuiGrid-root makeStyles-container__header-91 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <p
            class="MuiTypography-root makeStyles-item__header-92 MuiTypography-body2"
            data-testid="tableheader-lastModified"
          >
            DATE
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
        >
          <p
            class="MuiTypography-root makeStyles-item__header-92 MuiTypography-body2"
            data-testid="tableheader-content"
          >
            CONTENT
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-197 makeStyles-root-90"
      >
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            class="makeStyles-overflow-198"
            style="height: 700px; width: 200px; overflow: auto; position: relative; will-change: transform;"
          >
            <div
              style="height: 11900px; width: 100%;"
            >
              <div
                class="MuiGrid-root makeStyles-root-199 makeStyles-root-204 MuiGrid-container"
                data-testid="row-5e902a09035ac411cd04d22b"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-item__closeicon-201"
                  data-testid="row-closeicon-5e902a09035ac411cd04d22b"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                >
                  <div
                    class="MuiBox-root MuiBox-root-214 makeStyles-columns__item-202"
                  >
                    <p
                      class="MuiTypography-root makeStyles-columns__item__typo-203 MuiTypography-body2"
                    >
                      2020-04-10T08:10:49+00:00
                    </p>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                >
                  <div
                    class="MuiBox-root MuiBox-root-215 makeStyles-columns__item-202"
                  >
                    <p
                      class="MuiTypography-root makeStyles-columns__item__typo-203 MuiTypography-body2"
                    >
                      1212
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root makeStyles-root-199 makeStyles-root-216 MuiGrid-container"
                data-testid="row-5e6b34a80f14940526bb0780"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-item__closeicon-201"
                  data-testid="row-closeicon-5e6b34a80f14940526bb0780"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                >
                  <div
                    class="MuiBox-root MuiBox-root-217 makeStyles-columns__item-202"
                  >
                    <p
                      class="MuiTypography-root makeStyles-columns__item__typo-203 MuiTypography-body2"
                    >
                      2020-04-10T08:10:47+00:00
                    </p>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                >
                  <div
                    class="MuiBox-root MuiBox-root-218 makeStyles-columns__item-202"
                  >
                    <p
                      class="MuiTypography-root makeStyles-columns__item__typo-203 MuiTypography-body2"
                    >
                      13s123x232s sadaa
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root makeStyles-root-199 makeStyles-root-219 MuiGrid-container"
                data-testid="row-5e8a886b71b355002f81c196"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-item__closeicon-201"
                  data-testid="row-closeicon-5e8a886b71b355002f81c196"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                >
                  <div
                    class="MuiBox-root MuiBox-root-220 makeStyles-columns__item-202"
                  >
                    <p
                      class="MuiTypography-root makeStyles-columns__item__typo-203 MuiTypography-body2"
                    >
                      2020-04-06T01:39:55+00:00
                    </p>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                >
                  <div
                    class="MuiBox-root MuiBox-root-221 makeStyles-columns__item-202"
                  >
                    <p
                      class="MuiTypography-root makeStyles-columns__item__typo-203 MuiTypography-body2"
                    >
                      1212312
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 201px; height: 701px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
